lib/basic-auth-middleware.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const createError = require('http-errors');
+const debug = require('debug')('parkify:basic-auth-middleware');
+
+module.exports = function (request, response, next) {
+  debug('basic auth');
+
+  var authHeader = request.headers.authorization;
+  if (!authHeader) {
+    return next(createError(401, 'Authorization header required'));
+  }
+  var base64str = authHeader.split('Basic')[1];
```

Comment:
Needs a space after Basic, otherwise the it will have whitespace before the auth.